gdx-video-desktop/gdx-video-lwjgl3/src/com/badlogic/gdx/video/VideoPlayerDesktop.java

```diff
@@ -16,13 +16,16 @@
 
 package com.badlogic.gdx.video;
 
+import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.audio.Music;
+import com.badlogic.gdx.backends.lwjgl3.audio.mock.MockAudio;
 
 import java.nio.ByteBuffer;
 
 public class VideoPlayerDesktop extends CommonVideoPlayerDesktop {
 	@Override
 	Music createMusic (VideoDecoder decoder, ByteBuffer audioBuffer, int audioChannels, int sampleRate) {
+		if (Gdx.audio.getClass() == MockAudio.class) return null;
```

Comment:
We could also just return the MockAudio so we don't need the other null checks. Or am I missing something?

Reply:
That works in this scenario but doesn't provide protection against if a video is loaded without sound (either because it lacks an audio track or the decoding failed for whatever reason) in which case `audio` will be null or at least disposed (if the previous video it played had sound). And as `MockMusic` doesn't exist in the LWJGL2 backend, we can't create one in `CommonVideoPlayerDesktop` for no-audio videos.